File: ts-sdk/src/sdk/models/operations/postapireactionstoggle.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum PostApiReactionsToggleCategoryEnum {
    Like = "like",
    Readinglist = "readinglist",
    Unicorn = "unicorn"
}

export enum PostApiReactionsToggleReactableTypeEnum {
    Comment = "Comment",
    Article = "Article",
    User = "User"
}


export class PostApiReactionsToggleQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=category" })
  category: PostApiReactionsToggleCategoryEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=reactable_id" })
  reactableId: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=reactable_type" })
  reactableType: PostApiReactionsToggleReactableTypeEnum;
}


export class PostApiReactionsToggleRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: PostApiReactionsToggleQueryParams;
}


export class PostApiReactionsToggleResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: python-sdk/src/forem/models/operations/post_api_reactions_toggle.py
import dataclasses
from enum import Enum

class PostAPIReactionsToggleCategoryEnum(str, Enum):
    LIKE = "like"
    READINGLIST = "readinglist"
    UNICORN = "unicorn"

class PostAPIReactionsToggleReactableTypeEnum(str, Enum):
    COMMENT = "Comment"
    ARTICLE = "Article"
    USER = "User"


@dataclasses.dataclass
class PostAPIReactionsToggleQueryParams:
    category: PostAPIReactionsToggleCategoryEnum = dataclasses.field(metadata={'query_param': { 'field_name': 'category', 'style': 'form', 'explode': True }})
    reactable_id: int = dataclasses.field(metadata={'query_param': { 'field_name': 'reactable_id', 'style': 'form', 'explode': True }})
    reactable_type: PostAPIReactionsToggleReactableTypeEnum = dataclasses.field(metadata={'query_param': { 'field_name': 'reactable_type', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class PostAPIReactionsToggleRequest:
    query_params: PostAPIReactionsToggleQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class PostAPIReactionsToggleResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    
